### PR TITLE
coafile.rst: New section inheritance documentation

### PR DESCRIPTION
**Pull request by mridubhatnagar**
Add the reason for deprecating implicit inheritance
and instead using explicit inheritance.

Closes https://github.com/coala/documentation/issues/428

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by mridubhatnagar** [comment]
@Udayan12167  @Mixih hey can anyone of you please review this. It's not been updated since long. 
If changes seem fine. I will rebase it. Thanks

**Comment by Monal5031** [comment]
ack c1a8ad1

File: Users/coafile.rst [hunk 1]
```diff
@@ -85,7 +85,9 @@ This is the same file without section inheritance::
   other = some_other_value
 
 All settings must be part of a section, so don't do this for implicit
-inheritance (this is also deprecated behavior)::
+inheritance (this is also deprecated behavior). Implicit inheritance
+was leading to a section automatically getting inherited to all other
+sections without semantically making sense.::
```
Comment (sils) [inline Users/coafile.rst]: period and colon doesn't make sense

**Comment by sils** [comment]
unack https://github.com/coala/documentation/pull/439/commits/c1a8ad1015a0d36edf01d74e505f055fca205b13 the commit message lacks a verb

**Comment by sils** [comment]
sorry to spoil the fun ;)

**Comment by mridubhatnagar** [comment]
@sils thanks for pointing out those mistakes. I shall make changes soon :)

**Comment by mridubhatnagar** [comment]
@sils done with the changes you requested.

**Comment by shreyans800755** [comment]
ack c0420cf

**Comment by Makman2** [comment]
reack cf4ebcf

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**